apps/api/src/view-models/dynamicForms/buildFormFieldForCommandPayloadProp.ts

```diff
@@ -0,0 +1,97 @@
+import { FormFieldType, IFormField } from '@coscrad/api-interfaces';
+import {
+    ComplexCoscradDataType,
+    CoscradDataType,
+    CoscradPropertyTypeDefinition,
+    EnumTypeDefinition,
+    isSimpleCoscradPropertyTypeDefinition,
+} from '@coscrad/data-types';
+import { isNullOrUndefined } from '@coscrad/validation';
+import { InternalError } from '../../lib/errors/InternalError';
+import cloneToPlainObject from '../../lib/utilities/cloneToPlainObject';
+
+// TODO Move this to a higher level- possibly called CommandPayloadPropertyMeta
+type NameLabelAndDescription = {
+    name: string;
+    label: string;
+    description: string;
+};
+
+const lookupTable: { [K in CoscradDataType]: FormFieldType } = {
+    [CoscradDataType.NonEmptyString]: FormFieldType.textField,
+    [CoscradDataType.NonNegativeFiniteNumber]: FormFieldType.numericInput,
+    // Eventually, we want this to be a dynamic selection
+    [CoscradDataType.CompositeIdentifier]: FormFieldType.jsonInput,
+    [CoscradDataType.ISBN]: FormFieldType.textField,
+    [CoscradDataType.PositiveInteger]: FormFieldType.numericInput,
+    [CoscradDataType.RawData]: FormFieldType.jsonInput,
+    [CoscradDataType.URL]: FormFieldType.textField,
+    // Eventually, we want this to be a dynamic selection
+    [CoscradDataType.UUID]: FormFieldType.textField,
+    [CoscradDataType.Year]: FormFieldType.yearPicker,
+};
+
+const buildSimpleFormField = (
+    formFieldType: FormFieldType,
+    { name, label, description }: NameLabelAndDescription
+): IFormField => ({
+    type: formFieldType,
+    name,
+    label,
+    description,
+});
+
+/**
+ * TODO A factory pattern and polymorphic builders may be cleaner here
```

Comment:
Let's not  worry about cosmetics. Only do this if the current pattern raises an actual problem. Remove the TODO part of the comment.